src/rollback.rs

```diff
@@ -64,15 +64,48 @@ impl<K: Ord, V> AsRef<BTreeMap<K, V>> for RollbackableMap<K, V> {
     }
 }
 
-pub type RollbackableSet<T> = RollbackableMap<T, ()>;
+#[derive(Default)]
+pub struct RollbackableSet<K: Ord> {
+    map: BTreeMap<K, ()>,
+    old_entries: Vec<K>,
+}
 
 impl<T: Ord + Clone> RollbackableSet<T> {
-    pub fn add(&mut self, key: T) {
-        self.insert(key, ());
+    /// Adds `key` to the set and returns if it was already present.
+    pub fn add(&mut self, key: T) -> bool {
```

Comment:
Bikeshedding: Suggest to reverse the returned value (and maybe rename the method to `insert`) given that `BTreeSet` / `HashSet` return `true` from `insert` if the value is new.